backend/openai_interface/openai_interface.go

```diff
@@ -0,0 +1,67 @@
+package openai_interface
+
+import (
+	"context"
+	"errors"
+	"strings"
+
+	"github.com/sashabaranov/go-openai"
+)
+
+const IrrelevantQuery = "who is kim kardashian's husband?"
+const IrrelevantQueryFunctionalityIndicator = "If the input query has nothing to do with this prompt, return an empty string."
+const defaultQueryResponse = `{"query":"environment=production AND secure_session_id EXISTS","date_range":{"start_date":"","end_date":""}}`
+
+var MalformedPromptError = errors.New("empty or incorrect input query")
+
+type OpenAiInterface interface {
+	CreateChatCompletion(client *openai.Client, request openai.ChatCompletionRequest) (openai.ChatCompletionResponse, error)
+}
+
+type OpenAiImpl struct {
+}
+
+func (o *OpenAiImpl) CreateChatCompletion(client *openai.Client, request openai.ChatCompletionRequest) (openai.ChatCompletionResponse, error) {
+	if client == nil {
+		return openai.ChatCompletionResponse{}, errors.New("openai client is nil")
+	}
+	return client.CreateChatCompletion(context.Background(), request)
```

Comment:
pass the context thru from the caller, so the openAI inference can be traced back to the calling graphql operation